File: Examples/Chart/WinForm/VB.NET/ChartGallery/HighLow/NStandardHighLow2DUC.vb
Imports System
Imports System.Diagnostics
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms
Imports Nevron.Dom
Imports Nevron.GraphicsCore
Imports Nevron.Editors
Imports Nevron.Chart

Namespace Nevron.Examples.Chart.WinForm
	<ToolboxItem(False)> _
	Public Class NStandardHighLow2DUC
		Inherits NExampleBaseUC

		Private m_Chart As NChart
		Private m_HighLow As NHighLowSeries
		Private WithEvents DropLinesCheck As Nevron.UI.WinForm.Controls.NCheckBox
		Private WithEvents HighAreaFEButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents LowAreaFEButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents ShadowButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents MarkerStyleButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents LabelStyleButton As Nevron.UI.WinForm.Controls.NButton
		Private components As System.ComponentModel.Container = Nothing

		Public Sub New()
			InitializeComponent()
		End Sub

		''' <summary> 
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub


		#Region "Component Designer generated code"
		''' <summary> 
		''' Required method for Designer support - do not modify 
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.HighAreaFEButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.DropLinesCheck = New Nevron.UI.WinForm.Controls.NCheckBox()
			Me.LowAreaFEButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.ShadowButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.MarkerStyleButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.LabelStyleButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.SuspendLayout()
			' 
			' HighAreaFEButton
			' 
			Me.HighAreaFEButton.Location = New System.Drawing.Point(9, 9)
			Me.HighAreaFEButton.Name = "HighAreaFEButton"
			Me.HighAreaFEButton.Size = New System.Drawing.Size(161, 24)
			Me.HighAreaFEButton.TabIndex = 41
			Me.HighAreaFEButton.Text = "High Area Fill Style..."
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.HighAreaFEButton.Click += new System.EventHandler(this.AreaFEButton_Click);
			' 
			' DropLinesCheck
			' 
			Me.DropLinesCheck.ButtonProperties.BorderOffset = 2
			Me.DropLinesCheck.Location = New System.Drawing.Point(9, 166)
			Me.DropLinesCheck.Name = "DropLinesCheck"
			Me.DropLinesCheck.Size = New System.Drawing.Size(161, 21)
			Me.DropLinesCheck.TabIndex = 40
			Me.DropLinesCheck.Text = "Drop Lines"
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.DropLinesCheck.CheckedChanged += new System.EventHandler(this.DropLinesCheck_CheckedChanged);
			' 
			' LowAreaFEButton
			' 
			Me.LowAreaFEButton.Location = New System.Drawing.Point(9, 40)
			Me.LowAreaFEButton.Name = "LowAreaFEButton"
			Me.LowAreaFEButton.Size = New System.Drawing.Size(161, 24)
			Me.LowAreaFEButton.TabIndex = 0
			Me.LowAreaFEButton.Text = "Low Area Fill Style..."
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.LowAreaFEButton.Click += new System.EventHandler(this.LowAreaFEButton_Click);
			' 
			' ShadowButton
			' 
			Me.ShadowButton.Location = New System.Drawing.Point(9, 71)
			Me.ShadowButton.Name = "ShadowButton"
			Me.ShadowButton.Size = New System.Drawing.Size(161, 24)
			Me.ShadowButton.TabIndex = 50
			Me.ShadowButton.Text = "Shadow ..."
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.ShadowButton.Click += new System.EventHandler(this.ShadowButton_Click);
			' 
			' MarkerStyleButton
			' 
			Me.MarkerStyleButton.Location = New System.Drawing.Point(9, 102)
			Me.MarkerStyleButton.Name = "MarkerStyleButton"
			Me.MarkerStyleButton.Size = New System.Drawing.Size(161, 24)
			Me.MarkerStyleButton.TabIndex = 51
			Me.MarkerStyleButton.Text = "Marker Style..."
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.MarkerStyleButton.Click += new System.EventHandler(this.MarkerStyleButton_Click);
			' 
			' LabelStyleButton
			' 
			Me.LabelStyleButton.Location = New System.Drawing.Point(9, 133)
			Me.LabelStyleButton.Name = "LabelStyleButton"
			Me.LabelStyleButton.Size = New System.Drawing.Size(161, 24)
			Me.LabelStyleButton.TabIndex = 52
			Me.LabelStyleButton.Text = "Data Label Style..."
'INSTANT VB NOTE: The following InitializeComponent event wireup was converted to a 'Handles' clause:
'ORIGINAL LINE: this.LabelStyleButton.Click += new System.EventHandler(this.LabelStyleButton_Click);
			' 
			' NStandardHighLow2DUC
			' 
			Me.Controls.Add(Me.LabelStyleButton)
			Me.Controls.Add(Me.MarkerStyleButton)
			Me.Controls.Add(Me.ShadowButton)
			Me.Controls.Add(Me.HighAreaFEButton)
			Me.Controls.Add(Me.DropLinesCheck)
			Me.Controls.Add(Me.LowAreaFEButton)
			Me.Name = "NStandardHighLow2DUC"
			Me.Size = New System.Drawing.Size(180, 270)
			Me.ResumeLayout(False)

		End Sub
		#End Region

		Public Overrides Sub Initialize()
			MyBase.Initialize()

			' set a chart title
			Dim title As NLabel = nChartControl1.Labels.AddHeader("2D High Low Chart")
			title.TextStyle.FontStyle = New NFontStyle("Times New Roman", 18, FontStyle.Italic)
			title.TextStyle.FillStyle = New NColorFillStyle(GreyBlue)

			' configure the chart
			m_Chart = nChartControl1.Charts(0)
			m_Chart.Axis(StandardAxis.Depth).Visible = False

			' add a High-Low series
			m_HighLow = CType(m_Chart.Series.Add(SeriesType.HighLow), NHighLowSeries)
			m_HighLow.Name = "High-Low Series"
			m_HighLow.Legend.Mode = SeriesLegendMode.SeriesLogic
			m_HighLow.HighFillStyle = New NColorFillStyle(GreyBlue)
			m_HighLow.LowFillStyle = New NColorFillStyle(DarkOrange)
			m_HighLow.DataLabelStyle.Visible = False
			m_HighLow.DataLabelStyle.Format = "<high_value>:<low_value>"
			m_HighLow.LowValues.ValueFormatter = New NNumericValueFormatter("0.#")
			m_HighLow.HighValues.ValueFormatter = New NNumericValueFormatter("0.#")

			GenerateData()

			' apply layout
			ConfigureStandardLayout(m_Chart, title, nChartControl1.Legends(0))
		End Sub

		Private Sub GenerateData()
			m_HighLow.ClearDataPoints()

			For i As Integer = 0 To 19
				Dim d1 As Double = Math.Log(i + 1) + 0.1 * Random.NextDouble()
				Dim d2 As Double = d1 + Math.Cos(0.33 * i) + 0.1 * Random.NextDouble()

				m_HighLow.HighValues.Add(d1)
				m_HighLow.LowValues.Add(d2)
			Next i
		End Sub

		Private Sub AreaFEButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles HighAreaFEButton.Click
			Dim fillStyleResult As NFillStyle = Nothing

			If NFillStyleTypeEditor.Edit(m_HighLow.HighFillStyle, fillStyleResult) Then
				m_HighLow.HighFillStyle = fillStyleResult
				nChartControl1.Refresh()
			End If
		End Sub
		Private Sub LowAreaFEButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles LowAreaFEButton.Click
			Dim fillStyleResult As NFillStyle = Nothing

			If NFillStyleTypeEditor.Edit(m_HighLow.LowFillStyle, fillStyleResult) Then
				m_HighLow.LowFillStyle = fillStyleResult
				nChartControl1.Refresh()
			End If
		End Sub
		Private Sub ShadowButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles ShadowButton.Click
			Dim shadowStyleResult As NShadowStyle = Nothing

			If NShadowStyleTypeEditor.Edit(m_HighLow.ShadowStyle, shadowStyleResult) Then
				m_HighLow.ShadowStyle = shadowStyleResult
				nChartControl1.Refresh()
			End If
		End Sub
		Private Sub MarkerStyleButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles MarkerStyleButton.Click
			Dim markerStyleResult As NMarkerStyle = Nothing
			Dim series As NSeries = CType(nChartControl1.Charts(0).Series(0), NSeries)

			If NMarkerStyleTypeEditor.Edit(series.MarkerStyle, markerStyleResult) Then
				series.MarkerStyle = markerStyleResult
				nChartControl1.Refresh()
			End If
		End Sub
		Private Sub LabelStyleButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles LabelStyleButton.Click
			Dim styleResult As NDataLabelStyle = Nothing
			Dim series As NSeries = CType(nChartControl1.Charts(0).Series(0), NSeries)

			If NDataLabelStyleTypeEditor.Edit(series.DataLabelStyle, styleResult) Then
				series.DataLabelStyle = styleResult
				nChartControl1.Refresh()
			End If
		End Sub
		Private Sub DropLinesCheck_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles DropLinesCheck.CheckedChanged
			m_HighLow.DropLines = DropLinesCheck.Checked
			nChartControl1.Refresh()
		End Sub
	End Class
End Namespace

File: Examples/Chart/WinForm/CSharp/ChartGallery/HighLow/NStandardHighLow2DUC.cs
using System;
using System.Diagnostics;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;
using Nevron.Dom;
using Nevron.GraphicsCore;
using Nevron.Editors;
using Nevron.Chart;

namespace Nevron.Examples.Chart.WinForm
{
	[ToolboxItem(false)]
	public class NStandardHighLow2DUC : NExampleBaseUC
	{
		private NChart m_Chart;
		private NHighLowSeries m_HighLow;
		private Nevron.UI.WinForm.Controls.NCheckBox DropLinesCheck;
		private Nevron.UI.WinForm.Controls.NButton HighAreaFEButton;
		private Nevron.UI.WinForm.Controls.NButton LowAreaFEButton;
		private Nevron.UI.WinForm.Controls.NButton ShadowButton;
		private Nevron.UI.WinForm.Controls.NButton MarkerStyleButton;
		private Nevron.UI.WinForm.Controls.NButton LabelStyleButton;
		private System.ComponentModel.Container components = null;

		public NStandardHighLow2DUC()
		{
			InitializeComponent();
		}

		/// <summary> 
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(disposing);
		}


		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.HighAreaFEButton = new Nevron.UI.WinForm.Controls.NButton();
			this.DropLinesCheck = new Nevron.UI.WinForm.Controls.NCheckBox();
			this.LowAreaFEButton = new Nevron.UI.WinForm.Controls.NButton();
			this.ShadowButton = new Nevron.UI.WinForm.Controls.NButton();
			this.MarkerStyleButton = new Nevron.UI.WinForm.Controls.NButton();
			this.LabelStyleButton = new Nevron.UI.WinForm.Controls.NButton();
			this.SuspendLayout();
			// 
			// HighAreaFEButton
			// 
			this.HighAreaFEButton.Location = new System.Drawing.Point(9, 9);
			this.HighAreaFEButton.Name = "HighAreaFEButton";
			this.HighAreaFEButton.Size = new System.Drawing.Size(161, 24);
			this.HighAreaFEButton.TabIndex = 41;
			this.HighAreaFEButton.Text = "High Area Fill Style...";
			this.HighAreaFEButton.Click += new System.EventHandler(this.AreaFEButton_Click);
			// 
			// DropLinesCheck
			// 
			this.DropLinesCheck.ButtonProperties.BorderOffset = 2;
			this.DropLinesCheck.Location = new System.Drawing.Point(9, 166);
			this.DropLinesCheck.Name = "DropLinesCheck";
			this.DropLinesCheck.Size = new System.Drawing.Size(161, 21);
			this.DropLinesCheck.TabIndex = 40;
			this.DropLinesCheck.Text = "Drop Lines";
			this.DropLinesCheck.CheckedChanged += new System.EventHandler(this.DropLinesCheck_CheckedChanged);
			// 
			// LowAreaFEButton
			// 
			this.LowAreaFEButton.Location = new System.Drawing.Point(9, 40);
			this.LowAreaFEButton.Name = "LowAreaFEButton";
			this.LowAreaFEButton.Size = new System.Drawing.Size(161, 24);
			this.LowAreaFEButton.TabIndex = 0;
			this.LowAreaFEButton.Text = "Low Area Fill Style...";
			this.LowAreaFEButton.Click += new System.EventHandler(this.LowAreaFEButton_Click);
			// 
			// ShadowButton
			// 
			this.ShadowButton.Location = new System.Drawing.Point(9, 71);
			this.ShadowButton.Name = "ShadowButton";
			this.ShadowButton.Size = new System.Drawing.Size(161, 24);
			this.ShadowButton.TabIndex = 50;
			this.ShadowButton.Text = "Shadow ...";
			this.ShadowButton.Click += new System.EventHandler(this.ShadowButton_Click);
			// 
			// MarkerStyleButton
			// 
			this.MarkerStyleButton.Location = new System.Drawing.Point(9, 102);
			this.MarkerStyleButton.Name = "MarkerStyleButton";
			this.MarkerStyleButton.Size = new System.Drawing.Size(161, 24);
			this.MarkerStyleButton.TabIndex = 51;
			this.MarkerStyleButton.Text = "Marker Style...";
			this.MarkerStyleButton.Click += new System.EventHandler(this.MarkerStyleButton_Click);
			// 
			// LabelStyleButton
			// 
			this.LabelStyleButton.Location = new System.Drawing.Point(9, 133);
			this.LabelStyleButton.Name = "LabelStyleButton";
			this.LabelStyleButton.Size = new System.Drawing.Size(161, 24);
			this.LabelStyleButton.TabIndex = 52;
			this.LabelStyleButton.Text = "Data Label Style...";
			this.LabelStyleButton.Click += new System.EventHandler(this.LabelStyleButton_Click);
			// 
			// NStandardHighLow2DUC
			// 
			this.Controls.Add(this.LabelStyleButton);
			this.Controls.Add(this.MarkerStyleButton);
			this.Controls.Add(this.ShadowButton);
			this.Controls.Add(this.HighAreaFEButton);
			this.Controls.Add(this.DropLinesCheck);
			this.Controls.Add(this.LowAreaFEButton);
			this.Name = "NStandardHighLow2DUC";
			this.Size = new System.Drawing.Size(180, 270);
			this.ResumeLayout(false);

		}
		#endregion

		public override void Initialize()
		{
			base.Initialize();

			// set a chart title
			NLabel title = nChartControl1.Labels.AddHeader("2D High Low Chart");
			title.TextStyle.FontStyle = new NFontStyle("Times New Roman", 18, FontStyle.Italic);
			title.TextStyle.FillStyle = new NColorFillStyle(GreyBlue);

			// configure the chart
			m_Chart = nChartControl1.Charts[0];
			m_Chart.Axis(StandardAxis.Depth).Visible = false;

			// add a High-Low series
			m_HighLow = (NHighLowSeries)m_Chart.Series.Add(SeriesType.HighLow);
			m_HighLow.Name = "High-Low Series";
			m_HighLow.Legend.Mode = SeriesLegendMode.SeriesLogic;
			m_HighLow.HighFillStyle = new NColorFillStyle(GreyBlue);
			m_HighLow.LowFillStyle = new NColorFillStyle(DarkOrange);
			m_HighLow.DataLabelStyle.Visible = false;
			m_HighLow.DataLabelStyle.Format = "<high_value>:<low_value>";
			m_HighLow.LowValues.ValueFormatter = new NNumericValueFormatter("0.#");
			m_HighLow.HighValues.ValueFormatter = new NNumericValueFormatter("0.#");

			GenerateData();

			// apply layout
			ConfigureStandardLayout(m_Chart, title, nChartControl1.Legends[0]);
		}

		private void GenerateData()
		{
			m_HighLow.ClearDataPoints();

			for (int i = 0; i < 20; i++)
			{
				double d1 = Math.Log(i + 1) + 0.1 * Random.NextDouble();
				double d2 = d1 + Math.Cos(0.33 * i) + 0.1 * Random.NextDouble();

				m_HighLow.HighValues.Add(d1);
				m_HighLow.LowValues.Add(d2);
			}
		}

		private void AreaFEButton_Click(object sender, System.EventArgs e)
		{
			NFillStyle fillStyleResult;

			if (NFillStyleTypeEditor.Edit(m_HighLow.HighFillStyle, out fillStyleResult))
			{
				m_HighLow.HighFillStyle = fillStyleResult;
				nChartControl1.Refresh();
			}
		}
		private void LowAreaFEButton_Click(object sender, System.EventArgs e)
		{
			NFillStyle fillStyleResult;

			if (NFillStyleTypeEditor.Edit(m_HighLow.LowFillStyle, out fillStyleResult))
			{
				m_HighLow.LowFillStyle = fillStyleResult;
				nChartControl1.Refresh();
			}
		}
		private void ShadowButton_Click(object sender, System.EventArgs e)
		{
			NShadowStyle shadowStyleResult;

			if (NShadowStyleTypeEditor.Edit(m_HighLow.ShadowStyle, out shadowStyleResult))
			{
				m_HighLow.ShadowStyle = shadowStyleResult;
				nChartControl1.Refresh();
			}
		}
		private void MarkerStyleButton_Click(object sender, System.EventArgs e)
		{
			NMarkerStyle markerStyleResult;
			NSeries series = (NSeries)nChartControl1.Charts[0].Series[0];

			if (NMarkerStyleTypeEditor.Edit(series.MarkerStyle, out markerStyleResult))
			{
				series.MarkerStyle = markerStyleResult;
				nChartControl1.Refresh();
			}
		}
		private void LabelStyleButton_Click(object sender, System.EventArgs e)
		{
			NDataLabelStyle styleResult;
			NSeries series = (NSeries)nChartControl1.Charts[0].Series[0];

			if (NDataLabelStyleTypeEditor.Edit(series.DataLabelStyle, out styleResult))
			{
				series.DataLabelStyle = styleResult;
				nChartControl1.Refresh();
			}
		}
		private void DropLinesCheck_CheckedChanged(object sender, System.EventArgs e)
		{
			m_HighLow.DropLines = DropLinesCheck.Checked;
			nChartControl1.Refresh();
		}
	}
}